GH-3572: Migrate SFTP from `jsch` to `sshd-sftp` (#3892)
spring-projects/spring-integration: src/reference/asciidoc/sftp.adoc

Spring Integration supports sending and receiving files over SFTP by providing three client side endpoints: inbound channel adapter, outbound channel adapter, and outbound gateway.
It also provides convenient namespace configuration to define these client components.

> NOTE: Starting with version 6.0, an outdated JCraft JSch client has been replaced with modern https://mina.apache.org/sshd-project/index.html[Apache MINA SSHD] framework.

Reviewer: ```suggestion
NOTE: Starting with version 6.0, the outdated JCraft JSch client has been replaced with the https://mina.apache.org/sshd-project/index.html[Apache MINA SSHD] framework.
```
